python/ray/util/spark/cluster_init.py

```diff
@@ -935,16 +992,33 @@ def setup_ray_cluster(
             or referenced objects (either in-memory or spilled to disk). This parameter
             does not affect the head node.
             Default value is 1.0, minimum value is 0
-
+        is_global:
+            If True, set up and serve a global mode Ray on spark cluster.
+            The global Ray on spark cluster means:
+             - You can only create one active global Ray on spark cluster at a time.
+             - On databricks cluster, the global Ray cluster can be used by all users,
+               as contrast, non-global Ray cluster can only be used by current notebook
+               user.
+             - It is up persistently without automatic shutdown.
+             - On databricks notebook, you can connect to the global cluster by calling
+               ``ray.init()`` without specifying its address, it will discover the
+               global cluster automatically if it is up.
+
+            If global mode is enabled, calling `setup_ray_cluster` will be blocking,
+            once the call is interrupted, the global Ray on spark cluster is shut down.
+            For global mode, the ``ray_temp_root_dir`` argument is not supported. Global
+            model Ray cluster always use the default Ray temporary directory path.
     Returns:
-        A tuple of (address, remote_connection_address)
+        If ``is_global`` argument is False,
+        returns a tuple of (address, remote_connection_address)
         "address" is in format of "<ray_head_node_ip>:<port>"
         "remote_connection_address" is in format of
         "ray://<ray_head_node_ip>:<ray-client-server-port>",
         if your client runs on a machine that also hosts a Ray cluster node locally,
         you can connect to the Ray cluster via ``ray.init(address)``,
         otherwise you can connect to the Ray cluster via
         ``ray.init(remote_connection_address)``.
+        If ``is_global`` argument is True, returns None.
```

Comment:
why do we return None for global cluster?

Reply:
It is blocking mode, when it is interrupted, the cluster is shut down.

Reply:
https://github.com/ray-project/ray/pull/41153#discussion_r1406984298